internal/webhooks/clusterclass.go

```diff
@@ -234,9 +234,49 @@ func validateVariableUpdates(clusters []clusterv1.Cluster, oldClusterClass, newC
 				continue
 			}
 		}
+
+		if cluster.Spec.Topology.Workers == nil {
+			continue
+		}
+
+		for _, md := range cluster.Spec.Topology.Workers.MachineDeployments {
+			// Continue if there are no variable overrides.
+			if md.Variables == nil || len(md.Variables.Overrides) == 0 {
+				continue
+			}
+
+			for _, c := range md.Variables.Overrides {
+				// copy variable to avoid memory aliasing.
+				clusterVar := c
+
+				// 1) Error if a variable with a schema altered in the update is no longer valid on the Cluster.
+				if alteredVar, ok := varsDiff[variableValidationKey{clusterVar.Name, altered}]; ok {
+					if errs := variables.ValidateClusterVariable(&clusterVar, alteredVar, field.NewPath("")); len(errs) > 0 {
+						errorInfo.add(fmt.Sprintf("%s/%s", md.Name, alteredVar.Name), altered, cluster.Name)
+					}
+					continue
+				}
+
+				// 2) Error if a variable removed in the update is still in use in some Clusters.
```

Comment:
We could, but I"m not sure if we should. There is one difference which is the "name" of the variable. I'm not sure if it would make the code more complicated to read if I make that flexible in an extracted func.